lua/acf/server/sv_acfdamage.lua

```diff
@@ -241,32 +241,48 @@ end
 
 --Calculates the vector of the ricochet of a round upon impact at a set angle
 function ACF_RicochetVector(Flight, HitNormal)
-	--bit of maths shamelessly stolen from wiremod to rotate a vector around an axis
-	local Vec = Flight:GetNormalized()
-	local x,y,z = HitNormal[1], HitNormal[2], HitNormal[3]
-	
-	local length = (x*x+y*y+z*z)^0.5
-	x,y,z = x/length, y/length, z/length
-	local Rotated = -Vector((-1 + (x^2)*2) * Vec[1] + (x*y*2) * Vec[2] + (x*z*2) * Vec[3],
-	(y*x*2) * Vec[1] + (-1 + (y^2)*2) * Vec[2] + (y*z*2) * Vec[3],
-	(z*x*2) * Vec[1] + (z*y*2) * Vec[2] + (-1 + (z^2)*2) * Vec[3])
-	
-	return Rotated
+	local Vec = Flight:GetNormalized() 
+
+	return Vec - ( 2 * Vec:Dot(HitNormal) ) * HitNormal
 end
 
---Handles the impact of a round on a target
+-- Handles the impact of a round on a target
 function ACF_RoundImpact( Bullet, Speed, Energy, Target, HitPos, HitNormal , Bone  )
-	--if (Bullet.Type == "HEAT") then print("Pen: "..((Energy.Penetration / Bullet["PenAera"]) * ACF.KEtoRHA)) end
 	local Angle = ACF_GetHitAngle( HitNormal , Bullet["Flight"] )
+
+	local HitRes = ACF_Damage ( --DAMAGE !!
+		Target,
+		Energy,
+		Bullet["PenAera"],
+		Angle,
+		Bullet["Owner"],
+		Bone,
+		Bullet["Gun"],
+		Bullet["Type"]
+	)
+
 	local Ricochet = 0
-	local MinAngle = math.min(Bullet["Ricochet"] - Speed/39.37/18,89)	--Making the chance of a ricochet get higher as the speeds increase
-	if Angle > math.random(MinAngle,90) and Angle < 89.9 then	--Checking for ricochet
-		Ricochet = (Angle/100)			--If ricocheting, calculate how much of the energy is dumped into the plate and how much is carried by the ricochet
-		Energy.Penetration = Energy.Penetration - Energy.Penetration*Ricochet/5 --Ricocheting can save plates that would theorically get penetrated, can add up to 1/5 rating
+	if HitRes.Loss == 1 then
+		-- Ricochet distribution center
+		local sigmoidCenter = Bullet.DetonatorAngle or ( Bullet.Ricochet - math.abs(Speed / 39.37 - Bullet.LimitVel) / 100 )
+		
+		-- Ricochet probability (sigmoid distribution); up to 5% minimal ricochet probability for projectiles with caliber < 20 mm 
+		local ricoProb = math.Clamp( 1 / (1 + math.exp( (Angle - sigmoidCenter) / -4) ), math.max(-0.05 * (Bullet.Caliber - 2) / 2, 0), 1 )
+
+		-- Checking for ricochet
+		if ricoProb > math.random() and Angle < 90 then
+			Ricochet       = math.Clamp(Angle / 90, 0.05, 1) -- atleast 5% of energy is kept
+			HitRes.Loss    = 1 - Ricochet
+			Energy.Kinetic = Energy.Kinetic * HitRes.Loss

```

Comment:
Shouldn't this be `Energy.Kinetic = Energy.Kinetic * Ricochet`? With the current implementation, a 95% Loss will result in a 5% decrease in kinetic energy.